src/meili/traffic_segment_matcher.cc

```diff
@@ -122,11 +126,15 @@ namespace {
           merged.back()->end_percent_ = segment.end_percent_;
           merged.back()->ends_segment_ = segment.ends_segment_;
           merged.back().internal = merged.back().internal && is_internal(directed_edge);
+          // Turn off turn channel flag if any segment continuation is not a turn channel
+          if (merged.back().turn_channel && !is_turn_channel(directed_edge))
```

Comment:
yea, had it incorrect at first and then incorrectly thought this would not set it to false if any one in the continuation if false.